src/main/java/edu/escuelaing/ieti/matchmaking/services/impl/UserServiceImpl.java

```diff
@@ -57,6 +64,34 @@ public User getUserByEmail(String email) throws EntityNotFoundException {
         return optionalUser.orElseThrow(() -> new EntityNotFoundException(User.class, "User email", email));
     }
 
+    @Override
+    public List<User> getUserFriendsById(String userId) {
+        Optional<User> optionalUser = userRepository.findById(userId);
+        List<User> friends = new ArrayList<>();
+        optionalUser.ifPresent(user -> {
+            user.getFriends().forEach(friendId -> {
+                        userRepository.findById(friendId).ifPresent(friends::add);
+                    }
+            );
+        });
+        return friends;
+    }
+
+    @Override
+    public List<Team> getUserTeamsById(String userId) {
+        Optional<User> optionalUser = userRepository.findById(userId);
+        List<Team> teams = new ArrayList<>();
+        optionalUser.ifPresent(user -> {
+            user.getTeams().forEach(teamId -> {
```

Comment:
Instead of having a lambda here consider implementing a `map` operation in your stream to convert from Id to team object